cpp/src/io/parquet/compact_protocol_writer.cpp

```diff
@@ -298,6 +317,17 @@ inline void CompactProtocolFieldWriter::field_int_list(int field, std::vector<En
   current_field_value = field;
 }
 
+inline void CompactProtocolFieldWriter::field_int_list(int field, const std::vector<int64_t>& val)
+{
+  put_field_header(field, current_field_value, ST_FLD_LIST);
+  put_byte((uint8_t)((std::min(val.size(), (size_t)0xfu) << 4) | ST_FLD_I64));
+  if (val.size() >= 0xf) put_uint(val.size());
```

Comment:
Old code, but my guess is because `val.size()` is a `size_t`, then the other arg to `std::min` has to be as well. But it should be a `static_cast`. As for the next line, it's always a mystery to me what the compiler will complain about and what it will let slide...comparing an unsigned 64-bit int to a signed 32-bit int, you'd think there'd be a warning. 🤷‍♂️ I'll clean these up some.